src/testing/integration-tests/migration.md

```diff
@@ -0,0 +1,295 @@
+used in [widget testing][].
+
+For an introduction to the `integration_test` package, take a look at the
+[Integration Tests][] guide.
+
+## Starter example project
+
+The project in this guide is a small example desktop application with this
+functionality:
+
+* On the left, there's a list of plants that the user can scroll,
+tap and select.
+* On the right, there's a details screen that displays the plant name
+and species.
+* On app start, when no plant is selected, a text asking the user to select
+a plant is displayed
+* The list of plants is loaded from a local JSON file located in the
+assets folder.
+
+<img src='/assets/images/docs/integration-test/migration-1.png'
+class="mw-100" 
+alt="Starter project screenshot">
+
+You can find the full code example in the [Example Project][] folder.
+
+## Existing tests
+
+The project contains the three `flutter_driver` tests
+performing the following checks:
+
+* Verifying the initial status of the app.
+* Selecting the first item on the list of plants.
+* Scrolling and selecting the last item on the list of plants.
+
+The tests are contained in the `test_driver` folder, inside the `main_test.dart`
+file.
+
+In this folder there's also a file named `main.dart`, which contains a call to
+the method `enableFlutterDriverExtension()`. This file won't be necessary
+anymore when using `integration_test`.
+
+## Setup
+
+To start using the `integration_test` package add the `integration_test` to
+your pubspec.yaml file if you still haven't:
+
+```yaml
+dev_dependencies:
+  integration_test:
+    sdk: flutter
+```
+
+Next, in your project, create a new directory
+`integration_test/`, create your tests files there 
+with the format: `<name>_test.dart`.
+
+## Test migration
+
+This section contains different examples on how to migrate existing
+`flutter_driver` tests to `integration_test` tests.
+
+### Example: Verifying a widget is displayed
+
+When the app starts the screen on the left displays 
+a text asking the user to select one of the plants on the list.
+
+This test verifies that the text is displayed.
+
+**flutter_driver**
+
+In `flutter_driver`, the test uses `waitFor` which waits until the `finder`
+can locate the widget. The test will fail if the widget can't be found.
+
+<?code-excerpt "test_driver/main_test.dart (Test1)"?>
+```dart
+test('do not select any item, verify please select text is displayed',
+    () async {
+  // Wait for 'please select' text is displayed
+  await driver.waitFor(find.text('Please select a plant from the list.'));
+});
+```
+
+**integration_test**
+
+In `integration_test` you will have to perform two steps:
+
+First you have to load the main app widget using
+the `tester.pumpWidget` method.
+
+Then, you use `expect` with the matcher `findsOneWidget` to verify
+that the widget is displayed.
+
+<?code-excerpt "integration_test/main_test.dart (Test1)"?>
+```dart
+testWidgets('do not select any item, verify please select text is displayed',
```

Comment:
Here's an example of what I meant about using sentence-style output. (Ignore if this is against conventions, but by default we writers use Sentence-style for everything.)

```suggestion
testWidgets('Verify that the please select text is displayed. when no item is selected.',
```